app/controllers/api/base_controller.rb

```diff
@@ -260,6 +247,67 @@ def action_permission
       end
     end
 
+    def parent_permission(child_permission)
+      case child_permission.to_s
+      when 'create', 'destroy'
+        'edit'
+      when 'edit', 'view'
+        'view'
+      else
+        raise ::Foreman::Exception.new(N_("unknown parent permission for %s"), "#{params[:controller]}##{child_permission}")
+      end
+    end
+
+    def parent_resource_details
+      classes_arr = params.map do |key, value|
+        md = key.match(/(\w+)_id$/)
+        if md
+          values = [md[1], resource_class_for(resource_name_for(md[1])), value]
+        else
+          values = [nil, nil, nil]
+        end
+        values
+      end
+
+      parent_name, parent_class, parent_id = classes_arr.find{|kvp| not kvp[1].nil?}
+
+      return [nil, nil] if parent_name.nil?
+
+      parent_scope = scope_for(parent_class, :permission => "#{parent_permission(action_permission)}_#{parent_name.pluralize}")
+      parent_scope = select_by_resource_id_scope(parent_scope, parent_class, parent_id)
+      [parent_name, parent_scope]
+    end
+
+    #Should be replaced after moving to friendly_id version >= 5.0
+    def select_by_resource_id_scope(base_scope, resource_class, resource_id)
+      arel = resource_class.arel_table
```

Comment:
It's modified from the newer version: even the newer version does not support scopes.
Probably it would be a good idea to create a PR there that will enable scope functionality.